### ISSUE TITLE
Support a simple way to define exceptions for broad rules

### ISSUE THREAD

**Issue by na--** [ashanbrown/forbidigo#24]
There doesn't seem to be an easy way to specify a broad rule, but add a few exceptions to it. Specifically, I'd like to do the following:
1. Disable everything from the [`os` package](https://pkg.go.dev/os) everywhere across [k6](https://github.com/grafana/k6)
2. Add `//nolint:forbidigo` in [the single place](https://github.com/grafana/k6/blob/8da4149d1a597630cd2e2276263d2b6bdf7acd85/cmd/state/state.go#L61-L121) where we assemble our global state from the OS
3. Add exceptions for certain `os` types like [`os.Signal`](https://pkg.go.dev/os#Signal), which are safe to use and are valid return types even when we'd be mocking this global state in our tests.

Since Go [doesn't support negative lookaheads in regular expressions](https://stackoverflow.com/questions/26771592/negative-look-ahead-in-go-regular-expressions), I am not sure there is an easy way to currently do this? 

One way to solve this would be to be able to add a list (or even another regex) for every rule, for defining exceptions to the rule. 

Another way would be to specify that only matching a specific group in the regexp constitutes a linter error. Then we'd be able to use ["The Greatest Regex Trick Ever"](https://www.rexegg.com/regex-best-trick.html) to define rule exceptions :sweat_smile:

**Comment by ashanbrown** [comment]
@na-- thanks for sharing this use case.  Would it be possible to accomplish this by adding exceptions to `exclude-rules` in golangci-lint?

**Comment by na--** [comment]
Yeah, that should also work :thinking: From what I can see, the regular expressions there would have to be based on the whole error message and not just the Go identifiers, so it's a bit hacky and potentially fragile, but probably much less hacky than any other currently existing workarounds :sweat_smile:
